src/datepicker/datepicker.js

```diff
@@ -454,8 +459,11 @@ angular.module('mgcrea.ngStrap.datepicker',
 					controller.$setValidity('date', isValid);
 					controller.$setValidity('min', isMinValid);
 					controller.$setValidity('max', isMaxValid);
+
 					// Only update the model when we have a valid date
 					if (isValid) controller.$dateValue = parsedDate;
+
+					return isValid && isMaxValid && isMinValid;
```

Comment:
Changed this to return whether everything was valid so we can save the amount of calls to ```triggerInvalid()```